gatorgrouper/utils/group_creation.py

```diff
@@ -1,43 +1,85 @@
 """Contains all of the group creation algorithms"""
 
 import logging
-import itertools
 import random
+import itertools
 from typing import List, Union
 from gatorgrouper.utils import group_scoring
 
 
-# group_random.py
-def group_random_num_group(responses: str, numgrp: int) -> List[List[str]]:
+# pylint: disable=bad-continuation
+# pylint: disable=dangerous-default-value
+def group_random_group_size(
+    responses: str, grpsize: int, conflicts=[("foo", "bar", -1)]
+) -> List[List[str]]:
+    """ Calculate number of groups based on desired students per group """
+    # number of groups = number of students / minimum students per group
+    numgrp = int(len(responses) / grpsize)
+
+    return group_random_num_group(responses, numgrp, conflicts)
+
+
+# pylint: disable=dangerous-default-value
+# pylint: disable=too-many-locals
+def group_random_num_group(
+    responses: str, numgrp: int, conflicts=[("foo", "bar", -1)]
```

Comment:
Why are there default conflicts for a student named `foo` and `bar`?

Reply:
The concept is that this is an optional argument; that will not be used unless given a list in the form of a tuple `(str, str, int)`. The default will never be used since the argument is optional, but defaults are needed for the function to have the optional function call.

Reply:
The default should then be an empty list.

Reply:
Having the optional parameter not be force-formatted does not allow the code to compile as written, and is intentional. Specifying the default ensures that the given arguments follow the strict format of:
```
[(str1, str2, int1), (str3, str4, int2), ..., (strn-1, strn, intm)]
```
Because the parameter only matters in the event that the list of conflicts is given as an argument, and the default is generic in a way that it will never be an issue in the code (unless there is the off-chance that there are students whose full names are `foo` and `bar`, which is unfortunate), I believe this to be intentional and correct.

Reply:
A default argument does not enforce any typing or ensure any specific formatting -- any object would work, although might throw errors at runtime, since it is accessed as if it were a list. An empty list should be used in this case -- if it does not run, that is because there are other things wrong. If you want to document the format of `conflicts`, that should be done in the docstring, not code.

Reply:
Alright, I trust your judgment on this. I added a fix to the code that also allows the code to not crash by only checking the conflicts list if it is non-empty.